matrices.tex

```diff
@@ -25,6 +25,8 @@
 {x - sqrt(y*(x + y)): 1, x + sqrt(y*(x + y)): 1}
 \end{verbatim}
 
+%% TODO a Python list is not a dense representation...
+%% Is this supposed to say flat? - scopatz
```

Comment:
Well a list is collection of `PyObject*`, [namely](http://www.laurentluce.com/posts/python-list-implementation/), 

```c
typedef struct {
    PyObject_VAR_HEAD
    PyObject **ob_item;
    Py_ssize_t allocated;
} PyListObject;
```

The individual expression elements are all in their own locations on the heap. This is about as non-dense as it gets in terms of memory allocation. 

> It isn't really flat either. Does this help? 332203b

It isn't flat because it is a list of lists? That is fine too. I guess what I am trying to say is that the denseness and the data structure should be independent statements.  Something like, "The Matrix class represents dense matrices. Internally, these matrices store their elements as a list of lists."